snuba/utils/streams/consumer.py

```diff
@@ -516,6 +490,23 @@ def build_kafka_consumer_configuration(
     }
 
 
+@dataclass(frozen=True)
+class Commit:
+    __slots__ = ["group", "partition", "offset"]
+
+    group: str
+    partition: Partition
+    offset: int
+
+    def encode(self) -> Payload:
```

Comment:
Since we return a payload, calling this encode may be a bit too generic? What about `build_payload` ? No strong opinion though I would not block it for this.

Reply:
I agree that it is too generic (I didn't really intend for it to live here forever, it was just a convenient spot to put it), especially based on the previous comment. It should be moved to the `Codec` abstraction as long as you don't have any disagreement with that approach.